numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -451,6 +451,12 @@ typedef struct {
         int len;
 } PyArray_Dims;
 
+typedef enum PyNpCopyMode {
+        NPY_COPY_IF_NEEDED, 
+        NPY_COPY_ALWAYS, 
+        NPY_COPY_NEVER
+} PyArray_CopyMode;
```

Comment:
I don't think this has to be public API?  I also think you don't need to give the enum a name if you typedef it?